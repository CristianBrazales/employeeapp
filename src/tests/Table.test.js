import { render, screen } from "@testing-library/react";
import CustomTable from "../components/Table";
import renderer from "react-test-renderer";
test("renders our custom table no inputs", () => {
  render(<CustomTable />);
  const component = renderer.create(<CustomTable></CustomTable>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree.props.className).toEqual("container-table");
});

test("renders our custom table with data", () => {
  let data = [
    {
      email: "admin@admin.com",
      password: "$2a$10$MeZVYPCgw3RnHcjHebTaKepnl1y9Ma5v1q9y8avHiXwOTSxRGHYda",
      nombres: "cristian Brazales",
      cedula: "0023412312",
      apellidos: "Skywalker",
      roles: ["ADMIN", "USER"],
      id: 1,
      fechaNacimiento: "2022-05-31T04:00:00.000Z",
      direccion: "",
      telefono: "1242523",
      estadoVacunacion: "true",
      tipoVacuna: "Jhonson&Jhonson",
      fechaVacuna: "2022-06-14T05:37:09.000Z",
      numeroDosis: "1",
      username: "admin@admin.com",
      direccionDomicilio: "Death star av",
      correo: "admin@admin.com",
    },
  ];

  render(<CustomTable data={data}></CustomTable>);
  let testingNode;
  testingNode = screen.getByText(/^admin@admin.com/i);
  expect(testingNode).not.toBeNull();
});
