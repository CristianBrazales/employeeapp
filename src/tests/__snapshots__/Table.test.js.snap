// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders our custom table no inputs 1`] = `
<div
  className="container-table"
>
  <div
    className="table-responsive"
  >
    <table
      className="table table-sm table-bordered"
      style={
        Object {
          "margin": 0,
        }
      }
    >
      <thead>
        <tr>
          <th
            className="table-header-custom"
          >
            Cédula
          </th>
          <th
            className="table-header-custom"
          >
            Nombre
          </th>
          <th
            className="table-header-custom"
          >
            Apellido
          </th>
          <th
            className="table-header-custom"
          >
            Correo electrónico
          </th>
          <th
            className="table-header-custom"
          >
            Estado Vacunación
          </th>
          <th
            className="table-header-custom"
          >
            Fecha Vacunación
          </th>
          <th
            className="table-header-custom"
          >
            Tipo vacuna
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;
